app/lib/forums.js

```diff
@@ -870,69 +870,42 @@ var library = {
                     'JOIN mdl_user u ON u.id = log.userid ' +
                     'JOIN mdl_course c ON c.id = log.course ' +
                     'JOIN mdl_course_modules cm on cm.id = log.cmid ' +
-                    'JOIN mdl_forum_posts p ON p.id = log.info AND p.userid = log.userid ' +
-                    'JOIN mdl_forum_discussions d ON d.id = p.discussion AND d.course = log.course AND d.forum = cm.instance ' +
+                    'JOIN mdl_forum_posts p ON p.id = log.info ' +
+                    'JOIN mdl_forum_discussions d ON d.id = p.discussion ' +
                     'JOIN mdl_forum f ON f.id = cm.instance and f.course = log.course ' +
                     "WHERE log.module = 'forum' AND log.action = 'add post' AND " + restrict_clause,
 
         sql_match:  (row) => {
-            // Note: main reason for missed matches is duplicate user accounts, where the old
-            // one (whose ID would match here) has had their username+email renamed.
             return mysql.format(
                 'SELECT c.id AS course, ' +
                 '       u.id AS userid, u.username, u.email, ' +
                 '       cm.id AS cmid, ' +
-                '       d.id as did, d.name as discussion_name, ' +
```

Comment:
Many of the changes in forums.js seem to be removing fields that aren't needed for fn(). Those fields might come in handy when we're doing QA though, to make it easier to see that we've really matched the right row in the new DB.